modules/light-clients/07-tendermint/types/update_test.go

```diff
@@ -279,3 +280,66 @@ func (suite *TendermintTestSuite) TestCheckHeaderAndUpdateState() {
 		}
 	}
 }
+
+func (suite *TendermintTestSuite) TestPruneConsensusState() {
+	ctx := suite.chainA.GetContext().WithBlockTime(suite.now)
+
+	// create 2 consensus states with the clientTime along with the metadata that would be stored by UpdateClient
+	consensusState := types.NewConsensusState(suite.clientTime, commitmenttypes.NewMerkleRoot(suite.header.Header.GetAppHash()), suite.valsHash)
+	suite.chainA.App.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, clientID, height, consensusState)
+	types.SetProcessedTime(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), height, uint64(ctx.BlockTime().UnixNano()))
+	types.SetIterationKey(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), height)
+	nextHeight := height.Increment()
+	suite.chainA.App.IBCKeeper.ClientKeeper.SetClientConsensusState(ctx, clientID, nextHeight, consensusState)
+	types.SetProcessedTime(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), nextHeight, uint64(ctx.BlockTime().UnixNano()))
+	types.SetIterationKey(suite.chainA.App.IBCKeeper.ClientKeeper.ClientStore(ctx, clientID), nextHeight)
+
+	// Set consensus state to a timestamp a week from now so that it is valid for the next consensus state which will be two weeks from now.
```

Comment:
Why not just create a small trusting period?

Reply:
Decided not to do this because I want to test that an Update only prunes a single consensus state. This requires non uniform time intervals between updates. So i had to use `TimeIncrementBy` so no point changing the Trusting period anyway